### PR TITLE
refactor: move to @jest/globals

### PR DESCRIPTION
**Pull request by lemald**
Asana ticket: [⚙️ Switch to using `@jest/globals`](https://app.asana.com/0/1152340551558956/1204432665252071/f)

### CODE REVIEW

File: assets/tests/components/map/controls/layersControl.test.tsx [hunk 1]
```diff
@@ -14,7 +15,7 @@ const mapWrapper = ({ children }: { children: ReactNode }) => (
 
 describe("LayersControl", () => {
   test("when the control button is clicked, the list of layers is rendered with the selected layer marked as selected", async () => {
-    render(<LayersControl tileType="base" setTileType={jest.fn} />, {
+    render(<LayersControl tileType="base" setTileType={jest.fn()} />, {
```
Comment (firestack) [inline assets/tests/components/map/controls/layersControl.test.tsx:18]: ooo thanks for fixing this! I'm curious, did the new types help you catch this at all?

File: assets/tests/testHelpers/mockHelpers.ts [hunk 2]
```diff
@@ -80,12 +81,14 @@ export const mockFullStoryEvent = (): void => {
 }
 
 export const mockTileUrls = (): void => {
-  ;(tilesetUrlForType as jest.Mock).mockImplementation((type: TileType) => {
-    switch (type) {
-      case "base":
-        return "test_base_url/{z}/{x}/{y}"
-      case "satellite":
-        return "test_satellite_url/{z}/{x}/{y}"
+  ;(tilesetUrlForType as jest.Mock<(t: TileType) => string>).mockImplementation(
```
Comment (firestack) [inline assets/tests/testHelpers/mockHelpers.ts:84]: question: can this use `typeof`?
```suggestion
  ;(tilesetUrlForType as jest.Mock<typeof tilesetUrlForType>).mockImplementation(
```

File: assets/tests/models/routeTab.test.ts
Comment (firestack) [inline assets/tests/models/routeTab.test.ts:1]: __praise__: I like these refactor's/cleanups!

File: assets/tests/testHelpers/mockHelpers.ts (hunk 2 above)
Comment (lemald) [reply assets/tests/testHelpers/mockHelpers.ts:84]: Looks like it can!

File: assets/tests/testHelpers/socketHelpers.ts [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
   expectedReceiveData?: (() => any) | any
 ) => {
   const result = {
     join: jest.fn(),
     leave: jest.fn(),
-    on: jest.fn(),
-    receive: jest.fn(),
+    on: jest.fn<Channel["on"]>(),
+    receive:
+      jest.fn<(message: PushStatus, handler: (data?: any) => void) => void>(),
```
Comment (firestack) [inline assets/tests/testHelpers/socketHelpers.ts:18]: issue(non-blocking): it seems like the API we have here isn't quite right? From what I can tell, `receive` doesn't exist on the `Channel` type, but is instead on the `Push` type, which is returned by the `join`, `leave`, and `push` methods. I'm thinking we should open a tech debt ticket to fix this? 

I'd like to know what you think and if this is something we should revisit in the future.

File: assets/tests/components/map.test.tsx [hunk 4]
```diff
@@ -452,20 +466,23 @@ describe("<MapFollowingPrimaryVehicles />", () => {
 })
 
 describe("autoCenter", () => {
-  beforeEach(() => setHtmlDefaultWidthHeight(0, 0))
-  const Leaflet = jest.requireActual("leaflet")
+  beforeEach(() => {
+    setHtmlDefaultWidthHeight(0, 0)
+    return
+  })
```
Comment (firestack) [inline assets/tests/components/map.test.tsx:472]:
```suggestion
  beforeEach(() => {
    setHtmlDefaultWidthHeight(0, 0)
  })
```

File: assets/tests/testHelpers/socketHelpers.ts (hunk 3 above)
Comment (lemald) [reply assets/tests/testHelpers/socketHelpers.ts:18]: Ticket [here](https://app.asana.com/0/1152340551558956/1205307050135117/f).

**Review by firestack — APPROVED** [review]
LGTM!

**Pull request merged**